docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -36,6 +36,44 @@ Upgrades the contract code by updating its {class_hash}.
 
 NOTE: This function is usually protected by an xref:access.adoc[Access Control] mechanism.
 
+[.contract]
+[[IUpgradeAndCall]]
+=== `++IUpgradeAndCall++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.16.0/packages/upgrades/src/interface.cairo[{github-icon},role=heading-link]
+
+:Upgraded: xref:UpgradeableComponent-Upgraded[Upgraded]
+
+```cairo
+use openzeppelin::upgrades::interface::IUpgradeable;
+```
+
+Interface of an upgradeable contract that also offers the ability to upgrade and call a function in the upgraded context.
+
+[.contract-index]
+.Functions
+--
+* xref:#IUpgradeAndCall-upgrade[`++upgrade(new_class_hash)++`]
+* xref:#IUpgradeAndCall-upgrade_and_call[`++upgrade_and_call(new_class_hash, selector, calldata)++`]
+--
+
+[#IUpgradeAndCall-Functions]
+==== Functions
+
+[.contract-item]
+[[IUpgradeAndCall-upgrade]]
+==== `[.contract-item-name]#++upgrade++#++(new_class_hash: ClassHash)++` [.item-kind]#external#
```

Comment:
If we remove this from the interface let's not forget to remove the doc entry too.